### ISSUE TITLE
Same method name in difference services lead to error: "Multiple methods named 'send' are annotated with @ThriftMethod in the given services"

### ISSUE THREAD

**Issue by symfu** [airlift/drift#110]
Given to services:

@ThriftService
public interface ServiceOne {
    @ThriftMethod
    boolean sendMessage(String message);
}

@ThriftService
public interface ServiceTwo {
    @ThriftMethod
    boolean sendMessage(String message);
}

DriftServer will throw an exception complains that ""Multiple methods named '%s' are annotated with @ThriftMethod in the given services"".

It seems that difference services can't have a method with the same name? 

Relative code (io/airlift/drift/server/DriftServerMethodInvoker.java:58): 

`
package io.airlift.drift.server;

class DriftServerMethodInvoker implements ServerMethodInvoker {

    ...

    public DriftServerMethodInvoker(
            ThriftCodecManager codecManager,
            Collection<DriftService> services,
            List<MethodInvocationFilter> filters,
            MethodInvocationStatsFactory methodInvocationStatsFactory)
    {
        Map<String, ServiceMethod> processorMap = new HashMap<>();
        ImmutableMap.Builder<String, MethodInvocationStat> stats = ImmutableMap.builder();
        for (DriftService service : services) {
            ThriftServiceMetadata serviceMetadata = new ThriftServiceMetadata(service.getService().getClass(), codecManager.getCatalog());
            for (ThriftMethodMetadata thriftMethodMetadata : serviceMetadata.getMethods().values()) {
                if (processorMap.containsKey(thriftMethodMetadata.getName())) {

                    
                    throw new IllegalArgumentException(format("Multiple methods named '%s' are annotated with @ThriftMethod in the given services", thriftMethodMetadata.getName()));
--------------------^  HERE

                }
                ServiceMethod serviceMethod = new ServiceMethod(codecManager, service.getService(), thriftMethodMetadata, filters);
                processorMap.put(thriftMethodMetadata.getName(), serviceMethod);
                if (service.isStatsEnabled()) {
                    stats.put(thriftMethodMetadata.getName(), methodInvocationStatsFactory.getStat(serviceMetadata, service.getQualifier(), serviceMethod.getMethodMetadata()));
                }
            }
        }

        ...

    }
`

**Comment by electrum** [comment]
This is correct. Unfortunately, the Thrift protocol does not have any namespace for services. In fact, there’s no concept of “service” on this the wire. There is only the method name.

**Comment by hand515** [comment]
Thrift's TMultiplexedProtocol may be an ungraceful way to use the service name as namespace. It converts the method name to **serviceName:methodName** . And we can _@ThriftMethod(serviceName:methodName)_ to handle multiplex protocol.
